cheroot/ssl/pyopenssl.py

```diff
@@ -102,17 +102,103 @@ def _safe_call(self, is_reader, call, *args, **kwargs):
 
     def recv(self, size):
         """Receive message of a size from the socket."""
-        return self._safe_call(True, super(SSL_fileobject, self).recv, size)
+        return self._safe_call(
+            True,
+            super(SSLFileobjectStreamReader, self).recv,
+            size,
+        )
 
     def sendall(self, *args, **kwargs):
         """Send whole message to the socket."""
-        return self._safe_call(False, super(SSL_fileobject, self).sendall,
-                               *args, **kwargs)
+        return self._safe_call(
+            False,
+            super(SSLFileobjectStreamReader, self).sendall,
+            *args, **kwargs
+        )
 
     def send(self, *args, **kwargs):
         """Send some part of message to the socket."""
-        return self._safe_call(False, super(SSL_fileobject, self).send,
-                               *args, **kwargs)
+        return self._safe_call(
+            False,
+            super(SSLFileobjectStreamReader, self).send,
+            *args, **kwargs
+        )
+
+
+class SSLFileobjectStreamWriter(StreamWriter):
+    """SSL file object attached to a socket object."""
+
+    ssl_timeout = 3
+    ssl_retry = .01
+
+    def _safe_call(self, is_reader, call, *args, **kwargs):
```

Comment:
This function seems to add a lot of repetition. I suggest extracting the common aspects of _safe_call into a mix-in for each of these SSLFileobject classes.

Reply:
Yeah, I just did this quick and dirty, I'm also in favor of having a mixin. Just didn't get to actually doing it.